subsys/net/lib/sockets/sockets.c

```diff
@@ -375,3 +378,75 @@ int zsock_fcntl(int sock, int cmd, int flags)
 		return -1;
 	}
 }
+
+int zsock_poll(struct zsock_pollfd *fds, int nfds, int timeout)
+{
+	int i;
+	int ret = 0;
+	struct zsock_pollfd *pfd;
+	struct k_poll_event *pev;
+	struct k_poll_event *pev_end = poll_events + ARRAY_SIZE(poll_events);
+
+	if (timeout < 0) {
+		timeout = K_FOREVER;
+	}
+
+	pev = poll_events;
+	for (pfd = fds, i = nfds; i--; pfd++) {
+
+		/* Per POSIX, negative fd's are just ignored */
+		if (pfd->fd < 0) {
```

Comment:
It's the basic assumption of the current implementation that it won't be negative (i.e. that RAM lies in first half of the CPU address space), e.g. https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/lib/sockets/sockets.c#L78 . This fact will be documented in the docs (poll() is the last feature socket feature to make it feature-complete, so I'll work on docs in the next patches). If we face a case where RAM lies in 2nd half of address space, we can look how to deal with it. But I don't think we'll face such a case soon, at least I believe all boards currently supported by Zephyr have RAM in the 1st half.